packages/utils/telemetry-utils/src/batchedTelemetryHelper.ts

```diff
@@ -0,0 +1,92 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import type { IDisposable } from "@fluidframework/core-interfaces";
+
+import type {
+	ITelemetryGenericEventExt,
+	ITelemetryLoggerExt,
+	ITelemetryPerformanceEventExt,
+} from "./telemetryTypes.js";
+
+/**
+ * TODO
+ */
+interface ICustomDataMap {
+	[key: string]: number;
+}
+
+/**
+ * TODO
+ */
+export class BatchedTelemetryHelper implements IDisposable {
+	disposed: boolean = false;
+
+	// Stores value of the custom data passed into the logger.
+	private readonly customDataMap: Map<string, number> = new Map<string, number>();
+	// Counter to keep track of the number of times the log function is called.
+	private counter = 0;
+
+	/**
+	 * TODO
+	 */
+	public constructor(
+		private readonly eventBase: ITelemetryGenericEventExt,
+		private readonly logger: ITelemetryLoggerExt,
+		private readonly threshold: number,
+	) {}
+
+	private incrementThresholdCount(): void {
+		this.counter++;
+	}
+
+	private resetThresholdCount(): void {
+		this.counter = 0;
+	}
+
+	private isAboveThreshold(): boolean {
+		return this.counter >= this.threshold;
+	}
```

Comment:
Nit: they're only used once and they're one-line functions, I'd recommend we just do `this.counter++ / this.counter = 0 / this.counter >= this.threshold` inside the current callers of these. Calling a function is not super expensive, but it's also not _free_. In this case I think the overhead isn't worth it; the code is arguably just as readable (and still doesn't repeat itself) without the extra functions.